### PR TITLE
Restructure and add rust like conversion to and from

### LINKED ISSUE #30: Move de and ser into own folder
Restruct project structure.

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #30 
@oli-obk @RReverser r?

### CODE REVIEW

**Comment by farodin91** [comment]
Resolved!

File: src/ser/mod.rs [hunk 1]
```diff
@@ -44,11 +44,70 @@ pub fn serialize<W: Write, S: Serialize>(value: S, writer: W) -> Result<()> {
     to_writer(writer, &value)
 }
 
+
+/// A convenience method for serializing some object to a buffer.
+///
+/// You'll almost always want to use this over any of the other things in the
+/// `serde_xml_rs::to_writer` module.
```
Comment (oli-obk) [inline src/ser/mod.rs]: We have a `to_writer` module? 😆

**Comment by oli-obk** [comment]
Thanks for going above and beyond my request! Just a small nit (also on the `to_string` function).

**Comment by farodin91** [comment]
Should i decraepted serialize and deserialize

**Comment by oli-obk** [comment]
> Should i decraepted serialize and deserialize

This is a complete API breakage anyway, just remove them?

**Comment by farodin91** [comment]
Resolved

**Comment by oli-obk** [comment]
cc #27 (this improves the situation a lot imo)

**Pull request merged**